sympy/matrices/matrices.py

```diff
@@ -1583,11 +1583,11 @@ def _matrix_pow_by_recursion(self, num):
     def __pow__(self, num):
         if not self.is_square:
             raise NonSquareMatrixError()
-        if not sympify(num).is_integer and (num > 0) == True and not num % 1 == 0:
-            b = self.jordan_cells()[1]
-            for i in range(len(b)):
-                if (b[i]).rows > 1 and b[i][0, 0] == 0:
-                    raise ValueError("Cannot be evaluated")
+        num = sympify(num)
+        if not num.is_integer and (num >= 0) == True and not num % 1 == 0 and self.is_nilpotent():
```

Comment:
If the matrix is nilpotent, there is no need to call `self.eigenvals()` since 0 is the only eigenvalue.